src/bun.js/bindings/bindings.cpp

```diff
@@ -2455,6 +2455,48 @@ JSC__JSValue JSC__JSValue__fromEntries(JSC__JSGlobalObject* globalObject, ZigStr
     return JSC::JSValue::encode(object);
 }
 
+JSC__JSValue JSC__JSValue__keys(JSC__JSGlobalObject* globalObject, JSC__JSValue objectValue) {
+    JSC::VM& vm = globalObject->vm();
+
+    JSC::JSObject* object = JSC::JSValue::decode(objectValue).getObject();
+    JSC::JSType type = object->type();
+
+    switch(type){
+        // Map and Set always return empty array for Object.keys
```

Comment:
This isn't always true, a property added to a map/set after construction will appear in the array
<img width="192" alt="Screenshot 2024-01-01 at 8 32 13 AM" src="https://github.com/oven-sh/bun/assets/35280289/fce51d11-e451-4eb2-bc44-a64e7e0be09c">

Reply:
You are right 👍